### PR TITLE
chore: Remove ignores from import-linter

### PR DESCRIPTION
**Pull request by gruebel**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

- moved `doc_generator.py` to root
- `TerraformBlock` got his own `get_attribute_dict()` method and therefore removed TF related code from the 
base `Block` class
- changed GH token for danger job

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/pr-test.yml [hunk 1]
```diff
@@ -13,6 +13,13 @@ jobs:
 
   danger-check:
     runs-on: [ self-hosted, public, linux, x64 ]
+    permissions:
+      actions: write
```
Comment (james-otten-pan) [inline .github/workflows/pr-test.yml:17]: Why are the following required:

- actions
- checks
- contents
- statuses

**Review by ChanochShayner — APPROVED** [review]

File: .github/workflows/pr-test.yml [hunk 2]
```diff
@@ -22,7 +25,7 @@ jobs:
           node-version: "16"
       - name: Install and run DangerJS
         env:
-          GITHUB_TOKEN: ${{ secrets.DANGER_GITHUB_TOKEN }}
```
Comment (omryMen) [inline .github/workflows/pr-test.yml:25]: should we delete it now from somewhere ?

**Review by omryMen — APPROVED** [review]

File: checkov/terraform/graph_builder/graph_components/blocks.py [hunk 3]
```diff
@@ -91,6 +93,56 @@ def __eq__(self, other: object) -> bool:
             self.block_type == other.block_type and self.attributes == other.attributes and \
             self.id == other.id and self.has_dynamic_block == other.has_dynamic_block and self.source == other.source
 
+    def get_attribute_dict(self, add_hash: bool = True) -> Dict[str, Any]:
```
Comment (bo156) [inline checkov/terraform/graph_builder/graph_components/blocks.py:96]: maybe use `dict` instead of `Dict`?

**Review by bo156 — APPROVED** [review]

File: .github/workflows/pr-test.yml (hunk 2 above)
Comment (gruebel) [reply .github/workflows/pr-test.yml:25]: yeah, I will after merging 🙂

File: checkov/terraform/graph_builder/graph_components/blocks.py (hunk 3 above)
Comment (gruebel) [reply checkov/terraform/graph_builder/graph_components/blocks.py:96]: sure, just copy pasted it 😅

**Pull request merged**